controllers/tests/controller_server/addons_server_test.py

```diff
@@ -41,34 +34,90 @@ def setUp(self):
+        if replication_type == REPLICATION_TYPE_MIRROR:
+            self.request.parameters = {PARAMETERS_SYSTEM_ID: SYSTEM_ID,
+                                       PARAMETERS_COPY_TYPE: COPY_TYPE}
+            replication_request = ReplicationRequest(OBJECT_INTERNAL_ID, OTHER_OBJECT_INTERNAL_ID, SYSTEM_ID, COPY_TYPE,
+                                                     REPLICATION_TYPE_MIRROR)
+        else:
+            self.request.parameters = {PARAMETERS_REPLICATION_POLICY: replication_name}
+            replication_request = ReplicationRequest(OBJECT_INTERNAL_ID, None, None, REPLICATION_COPY_TYPE_SYNC,
+                                                     REPLICATION_TYPE_EAR, replication_name)
+        return replication_request
+
+
+class TestControllerServicerEnableVolumeReplication(BaseControllerServicerSetUp, CommonControllerTest):
+    @property
+    def tested_method(self):
+        return self.servicer.EnableVolumeReplication
+
+    @property
+    def tested_method_response_class(self):
+        return pb2.EnableVolumeReplicationResponse
+
+    def setUp(self):
+        super().setUp()
         self.mediator.get_object_by_id = Mock()
         self.mediator.get_object_by_id.return_value = utils.get_mock_mediator_response_volume(10, VOLUME_NAME,
                                                                                               VOLUME_UID, "xiv")
-        self.mediator.get_replication = Mock()
-        replication_mock = utils.get_mock_mediator_response_replication(REPLICATION_NAME,
-                                                                        OBJECT_INTERNAL_ID,
-                                                                        OTHER_OBJECT_INTERNAL_ID)
-        self.mediator.get_replication.return_value = replication_mock
-        self.context = utils.FakeContext()
 
-    def _prepare_enable_replication_mocks(self):
-        self.mediator.get_replication = Mock()
-        self.mediator.get_replication.return_value = None
-        self.mediator.create_replication = Mock()
-
-    def test_enable_replication_succeeds(self):
-        self._prepare_enable_replication_mocks()
+    def _test_enable_replication_succeeds(self, replication_type):
+        replication_request = self._prepare_request_params(replication_type)
+        self._prepare_replication_mocks()
 
         self.servicer.EnableVolumeReplication(self.request, self.context)
 
         self.assertEqual(grpc.StatusCode.OK, self.context.code)
-        replication_request = ReplicationRequest(OBJECT_INTERNAL_ID, OTHER_OBJECT_INTERNAL_ID, SYSTEM_ID, COPY_TYPE,
-                                                 REPLICATION_TYPE_MIRROR)
         self.mediator.get_replication.assert_called_once_with(replication_request)
         self.mediator.create_replication.assert_called_once_with(replication_request)
 
+    def _test_enable_replication_no_volume_fails(self):
```

Comment:
I am not sure but by [this](https://github.com/IBM/ibm-block-csi-driver/blob/6213d0a4386b22cdfbee3ebf8113d141be7691b1/controllers/servers/csi/addons_server.py#L32) line, I think it raises an error, if this true, can you add an assert for exception `ObjectNotFoundError`?

Reply:
my bad